elyra/pipeline/component_registry.py

```diff
@@ -62,22 +79,22 @@ def __init__(self, component_registry_location: str, parser: ComponentParser):
     def registry_location(self) -> str:
         return self._component_registry_location
 
-    def get_all_components(self) -> List[Component]:
+    def get_all_components(self) -> Tuple[List[Component], List[Dict]]:
```

Comment:
What is being returned here?  Is this really the best design to implement?

Reply:
> Is this really the best design to implement?

This seems a little harsh - no?  Here's why this was changed to a tuple is due to the late-breaking discovery of a malformed palette that existed since custom components were introduced into master a few weeks ago: https://github.com/elyra-ai/elyra/pull/1882#discussion_r673228696  

However, I didn't fully convey how best to implement this workaround.  Now, after looking closer, the solution will be more involved due to the additional parsing that is required for components and the "adjusted_id" - that appears to only apply to airflow components. That said, since only the `CachedComponentRegistry` is used in this "design", the callers get the correct result despite the tuple.  I think what _should_ happen is the parser code  necessary only in `get_all_components()` but not `get_all_categories()` should be moved into the read method so that it returns "final results".  Then the base `ComponentRegistry` will just continue reading the file, once for each call to `get_all_components()` and `get_all_categories()`, while the `CachedComponentRegistry` will call the read method directly so that the file only gets read once (every 60 seconds).  What is causing trouble with that "workaround" is with the way `get_component()` works with its need to handle `adjusted_id`.  

I do have a question: do we also need to handle `adjusted_id` in `get_all_components()`?

This is worth further discussion, preferably via a virtual call.

Reply:
Sorry, I didn't mean to be harsh, but the API seemed a little strange, and the unpacking requirement... wanted basically to start a discussion and agree we could handle it on a call (maybe as an item on the scrum one).